File: backend/IDManager.Api/Domain/Entities/PointTransaction.cs
using IDManager.Api.Domain.Enums;

namespace IDManager.Api.Domain.Entities;

public class PointTransaction
{
    public int Id { get; set; }
    public int UserId { get; set; }
    public int ByUserId { get; set; }
    public int Points { get; set; }
    public PointTransType Type { get; set; }
    public PointStatus Status { get; set; } = PointStatus.Completed;
    public string Reason { get; set; } = string.Empty;
    public int? ForIdCardId { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: frontend/idmanager_app/lib/core_engine/points/domain/point_transaction.dart
import 'package:freezed_annotation/freezed_annotation.dart';

import '../../common/enums.dart';

part 'point_transaction.freezed.dart';

@freezed
sealed class PointTransaction with _$PointTransaction {
  const factory PointTransaction({
    required DateTime date,
    required String description,
    required int points,
    required PointTransType type,
    required PointStatus status,
  }) = _PointTransaction;
}
